components/buildList/js/buildList.controller.js

```diff
@@ -6,6 +6,464 @@ buildApp.controller('buildListController', ['$scope', '$q', '$interval', 'buildL
+				fStates.push (firewall.status);
+				angular.forEach(firewall.jobs, function(job, jobName) {
+					buildJobs[jobName].percentage += job.percentage;
+					buildJobs[jobName].status.push(job.status);
+				});
+			});
+
+			angular.forEach(buildJobs, function(value) {
+				value.percentage = parseInt(value.percentage / build.firewalls.length, 10);
+				value.status = setStatusOfProcess(value.status);
+			});
+
+			build.jobs = buildJobs;
+			build.status = setStatusOfProcess(fStates);
+			buildDefered.resolve();
+		});
+	}
+
+
+	/**
+	 * Starts to calculate the states of the builds.
+	 * @method updateBuildsList
+	 */
+	function updateBuildsList() {
+
+		timeElapsed += intervalTime;
+		var overallStatus = 'running';
+		var buildPromises = [];
+
+		angular.forEach($scope.buildList, function(build) {
+			var buildDefered = $q.defer();
+			buildPromises.push(buildDefered.promise);
+			updateBuild(build, buildDefered);
+		});
+
+		$q.all(buildPromises).then(function() {
+			var buildStates = [];
+
+			angular.forEach($scope.buildList, function(build) {
+				buildStates.push (build.status);
+			});
+
+			overallStatus = setStatusOfProcess(buildStates);
+			if (overallStatus === 'complete' || overallStatus === 'rejected') {
+				$interval.cancel(updateBuildsInterval);
+			}
+		});
+	}
+
+	init();
+}]).controller('buildListController', ['$scope', '$q', '$interval', 'buildListService', function ($scope, $q, $interval, buildListService) {
+	
+	var deferred = $q.defer(),
+		updateBuildsInterval,
+		intervalTime = 5000,
+		timeElapsed = 0;
+
+	/**
+	 * Initializes the functionality of this controller
+	 *  - requests the data
+	 *  - starts to calculate all the processes
+	 * @method init
+	 */
+	function init() {
+		getData();
+		$q.when(deferred.promise, function() {
+			updateBuildsInterval = $interval(updateBuildsList, intervalTime);
+		});
+	}
+
+	/**
+	 * Requests for the build list data
+	 * @method getData
+	 */
+	function getData() {
+		return buildListService.getData().then(function(response) { 
+			$scope.buildList = response.data;
+			deferred.resolve();
+		});
+	}
+
+	/**
+	 * This is called when details about a specific build or firewall are requested
+	 *  - if the build status is not complete or rejected then we can't display the results
+	 * @method displayResults
+	 * @param  {Object} item Item for which details are requested
+	 * @return {Boolean} if status of item is not complete or rejected, false value is returned otherwise true
+	 */
+	$scope.displayResults = function(item) {
+		if (item.status !== 'complete' && item.status !== 'rejected') {
+			return false;
+		}
+
+		buildListService.getResults(item.id).then(function(response) {
+			item.results = response.data;
+		});
+
+		return true;
+	};
+
+	/**
+	 * Establishes the status of a specific job (metrics, build, unit test, functional test)
+	 * - if the completness percentage it's greater or equal with 100% then a request to the server 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3ATrailingWhitespace)